bin/nyc.js

```diff
@@ -36,21 +34,60 @@ if ([
   }
   if (argv.all) nyc.addAllFiles()
 
-  var env = {
-    // Support running nyc as a user without HOME (e.g. linux 'nobody'),
-    // https://github.com/istanbuljs/nyc/issues/951
-    SPAWN_WRAP_SHIM_ROOT: process.env.SPAWN_WRAP_SHIM_ROOT || process.env.XDG_CACHE_HOME || require('os').homedir(),
-    NYC_CONFIG: JSON.stringify(argv),
-    NYC_CWD: process.cwd()
-  }
+  if (argv.setNodeOptions) {
+    const wrapper = require.resolve('../lib/wrap.js')
```

Comment:
The idea with `node-preload` is that we would replace this conditional block with the following:
```js
const nodePreload = require('node-preload')
nodePreload.preloadInsert(require.resolve('../lib/wrap.js'))
nodePreload.propagateEnv('NYC_CWD', process.cwd())
nodePreload.propagateEnv('NYC_CONFIG', JSON.stringify(argv))
if (argv['babel-cache'] === false) {
  nodePreload.propagateEnv('BABEL_DISABLE_CACHE', '1');
  process.env.BABEL_DISABLE_CACHE = 1;
}
```

Obviously I welcome comments/suggestions on improving the API of this.